src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -164,7 +165,20 @@ + (BOOL)supportsSecureCoding
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"<MTRDeviceClusterData: dataVersion %@>", _dataVersion];
+    return [NSString stringWithFormat:@"<MTRDeviceClusterData: dataVersion %@ attributes count %lu>", _dataVersion, static_cast<unsigned long>(_attributes.count)];
+}
+
+- (nullable instancetype)initWithDataVersion:(NSNumber * _Nullable)dataVersion attributes:(NSDictionary<NSNumber *, MTRDeviceDataValueDictionary> * _Nullable)attributes
```

Comment:
Please document that the number is the attribute id.

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/32778